api/app/tests/apps/services/domain/test_auto_create_children.py

```diff
@@ -0,0 +1,234 @@
+                                      category_assignment__category=self.category_container_is_vol)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+    def test_one_container_selected(self):
+        extra_properties = copy.deepcopy(self.extra_properties)
+        extra_properties[0]['answer'] = [
+            {'id': 'TEST-1', 'type': 'Rest', 'description': 'Restafval container'}
+        ]
+
+        signal = SignalFactory.create(extra_properties=extra_properties,
+                                      category_assignment__category=self.category_container_is_vol)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+    def test_too_many_containers_selected(self):
+        extra_properties = copy.deepcopy(self.extra_properties)
+        extra_properties[0]['answer'] = []
+
+        for x in range(25):
+            extra_properties[0]['answer'].append(
+                {'id': f'TEST-{x}', 'type': 'Rest', 'description': 'Restafval container'}
+            )
+
+        signal = SignalFactory.create(extra_properties=extra_properties,
+                                      category_assignment__category=self.category_container_is_vol)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+    def test_wrong_category(self):
+        extra_properties = copy.deepcopy(self.extra_properties)
+        extra_properties[0]['category_url'] = self.category_overig.get_absolute_url()
+
+        signal = SignalFactory.create(extra_properties=extra_properties,
+                                      category_assignment__category=self.category_overig)
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+        AutoCreateChildrenService.run(signal_id=signal.pk)
+
+        signal.refresh_from_db()
+
+        self.assertFalse(signal.is_parent)
+        self.assertEqual(signal.children.count(), 0)
+
+    def test_signal_wrong_state(self):
+        extra_properties = copy.deepcopy(self.extra_properties)
+        signal = SignalFactory.create(extra_properties=extra_properties,
+                                      category_assignment__category=self.category_container_is_vol,
+                                      status__state='i')
```

Comment:
Small, non-blocker, comment. We could set the correct status explicitly for the other tests in this class (so that it is clear we want status GEMELD 'm').